explorer/interpreter/action_stack.cpp

```diff
@@ -155,16 +165,37 @@ auto ActionStack::FinishAction(Nonnull<const Value*> result)
     case Action::Kind::ExpressionAction:
     case Action::Kind::LValAction:
     case Action::Kind::PatternAction:
-      PopScopes();
+      destroy_actions_ = PopScopes();
       SetResult(result);
+      break;
+    case Action::Kind::CleanUpAction:
+      break;
+  }
+  auto& scope = act->scope();
```

Comment:
Note the duplication here with the other FinishAction -- please use a helper function if this is needed. But it may also be worth checking how FinishAction is called, to see if it belongs here or at the call site.

Reply:
Finished!